source/Random.cpp

```diff
@@ -113,3 +113,29 @@ double Random::Normal()
 #endif
 	return normal(gen);
 }
+
+
+
+// Return a number from [0,1] derived from a normal curve,
+// compressed according to smoothness.
+double Random::CompressedNormal(double smoothness)
+{
+	double randomFactor = Normal();
+	// Compress values above and below the mean into [0, 1].
+	randomFactor = (randomFactor + smoothness) / (2 * smoothness);
+	// Retain only the fractional information, creating redundancy.
+	// Might be possible to get away with int32_t here, not sure.
+	randomFactor = randomFactor - static_cast<int64_t>(randomFactor);
+	// Push negative values into the usable range.
+	if(randomFactor < 0)
+		randomFactor++;
+
+	// Negative smoothness concentrates output toward 0 and 1.
+	if(smoothness < 0)
+	{
+		randomFactor += 0.5;
+		randomFactor = randomFactor - static_cast<int32_t>(randomFactor);
+	}
+	
+	return randomFactor;
```

Comment:
> So what I think is the goal here is to transform a randomly generated input into a value on the range `[a, b)` such that the values in the given range have a specific distribution: https://en.m.wikipedia.org/wiki/Inverse_transform_sampling

I'll have to investigate this further but it seems like it would abandon my currently functioning logic.  Hopefully whatever result that I can produce using it will be cheaper.

> To echo your own statement:
> 
> > On small inaccuracies or without many examples to compare at once (many submunitions) you're not really going to notice anything regardless of distribution as long as both halves of the inaccuracy (inaccurate left and inaccurate right) are equal in chance.
> 
> So we don't need to offer infinite choices and can instead offer preselected shapes that will be observably distinct (i.e. separate `Distributions` values).

Yes, I have to agree with you on this.  I'm working on consolidating the logic and distributions into their own class (meaning Projectile would lose its static method, but we would have another class hanging around) with 3 shapes for normal spread (tight, middling, wide).  Their inverses (projectiles gathered at endpoints) can be included trivially.  If you would prefer that I don't push everything to do with inaccuracy into its own class I certainly don't have to.  Also, this may seem like an about-face from me because I hadn't been keen on this idea, but Projectile and Random aren't good places for the logic and the new enums have to go somewhere.  Why not put it all together?

> To summarize, you want to pass a value from `Random::Real()` (our uniform distribution over `[0, 1)`) into a `Distribution`-specific function that maps it into `[-1, +1)` by inverting the cumulative distribution function for your desired probability distribution. Then in the projectile::inaccuracy method, you can select the appropriate one and multiply the value by the magnitude of the inaccuracy.

As mentioned above, it would become `Inaccuracy::Generate()` or something similar when I complete the new class and successfully attach it to Weapon (and Hardpoint and Projectile, nominally).  Using `[-1, +1]` is actually a better idea than the current `[0, 2] - 1` approach.  I don't know if `Random::Angle()` could be updated or overloaded with that logic for generating uniform distributions.  Don't think it would work for triangular inaccuracy.

Reply:
It seems to me that inverse transform sampling doesn't apply well to the normal distribution.  R seems to package the approximations with (this code)[https://github.com/SurajGupta/r-source/blob/master/src/nmath/qnorm.c], though it comes with dependencies. and probably wouldn't be easy to add.

I'm going to benchmark random normal number generation using the Box-Muller transform against the current method.  I'm not really certain if the 2-value generation can be fully taken advantage of by a static class (Random) but I'm thinking it's probably fine because I think it's already doing it with `gen`?

Reply:
According to my tests over 100 million iterations, the Box-Muller method generates normally distributed values 15% faster (takes 0.87x the time) than std::normal_distribution does.  It is also sufficiently accurate compared to the std method.  This saves 11.5% (takes 0.90x the time) at a minimum, assuming we do stick with my original choice to use normally distributed values only.  (For others your suggestion of inverse transformation would likely apply.)